internal/dao/services/event_processing/sync_install_dao_contracts.go

```diff
@@ -0,0 +1,67 @@
+package event_processing
+
+import (
+	"context"
+
+	"github.com/make-software/casper-go-sdk/sse"
+	"go.uber.org/zap"
+
+	"casper-dao-middleware/internal/dao/di"
+)
+
+type SyncInstallDAOContracts struct {
+	di.EntityManagerAware
+	di.CasperClientAware
+	di.DAOContractsMetadataAware
+	di.CESParserAware
+
+	daoContractsInstallBlocks []uint64
+}
+
+func NewSyncInstallDAOContracts() SyncInstallDAOContracts {
+	return SyncInstallDAOContracts{}
+}
+
+func (c *SyncInstallDAOContracts) SetDAOContractsInstallBlocks(heights []uint64) {
+	c.daoContractsInstallBlocks = heights
+}
+
+func (c *SyncInstallDAOContracts) Execute() error {
+	for _, blockHeight := range c.daoContractsInstallBlocks {
```

Comment:
Here instead of the range we could start from the first block and increment the height until we reach a non-existing block